src/core/map/coord.rs

```diff
@@ -474,28 +474,31 @@ mod coord_tests {
 	}
 
 	#[test]
-	fn coord_orientation_ring_iterator_count() {
+	fn coord_orientation_ring_iterator_small_count() {
 		{
 			let mut iter = CoordOrientationRingIterator::new(0);
 			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(0, 0)));
 			assert_eq!(iter.next(), None);
 		}
 		{
-			let mut iter = CoordOrientationRingIterator::new(1);
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(1, 0)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(0, 1)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(-1, 1)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(-1, 0)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(0, -1)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(1, -1)));
-			assert_eq!(iter.next(), None);
+			let coords =
+				CoordOrientationRingIterator::new(1).collect::<HashSet<CoordOrientation>>();
+			assert!(coords.contains(&CoordOrientation::new_axial(1, 0)));
+			assert!(coords.contains(&CoordOrientation::new_axial(0, 1)));
+			assert!(coords.contains(&CoordOrientation::new_axial(-1, 1)));
+			assert!(coords.contains(&CoordOrientation::new_axial(-1, 0)));
+			assert!(coords.contains(&CoordOrientation::new_axial(0, -1)));
+			assert!(coords.contains(&CoordOrientation::new_axial(1, -1)));
+			assert_eq!(coords.len(), 6);
```

Comment:
Good change, the order is an implementation detail and so shouldn't be checked, a `HashSet` is much better.  :-)